test/e2e/image_sign_test.go

```diff
@@ -0,0 +1,63 @@
+// +build !remote
+
+package integration
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	. "github.com/containers/podman/v2/test/utils"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Podman image sign", func() {
+	var (
+		tempdir    string
+		err        error
+		podmanTest *PodmanTestIntegration
+	)
+
+	BeforeEach(func() {
+		tempdir, err = CreateTempDirInTempDir()
+		if err != nil {
+			os.Exit(1)
+		}
+		podmanTest = PodmanTestCreate(tempdir)
+		podmanTest.Setup()
+		podmanTest.SeedImages()
+
+		tempGUNPGHOME := filepath.Join(podmanTest.TempDir, "tmpGPG")
+		err := os.Mkdir(tempGUNPGHOME, os.ModePerm)
+		Expect(err).To(Equal(nil))
+
+		origGNUPGHOME := os.Getenv("GNUPGHOME")
+		defer os.Setenv("GNUPGHOME", origGNUPGHOME)
+		err = os.Setenv("GNUPGHOME", tempGUNPGHOME)
+		Expect(err).To(Equal(nil))
+
+	})
+
+	AfterEach(func() {
+		podmanTest.Cleanup()
+		f := CurrentGinkgoTestDescription()
+		processTestResult(f)
+
+	})
+
+	It("podman sign image", func() {
+		cmd := exec.Command("gpg", "--import", "sign/secret-key.asc")
+		err := cmd.Run()
+		Expect(err).To(BeNil())
+		sigDir := filepath.Join(podmanTest.TempDir, "test-sign")
+		err = os.MkdirAll(sigDir, os.ModePerm)
+		Expect(err).To(BeNil())
+		session := podmanTest.Podman([]string{"image", "sign", "--directory", sigDir, "--sign-by", "foo@bar.com", "docker://library/alpine"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		sigDigestDir := "alpine@sha256=185518070891758909c9f839cf4ca393ee977ac378609f700f60a771a2dfe321"
```

Comment:
Does this work long-term? Looking at `CACHE_IMAGES`, `alpine:latest` is pulled from an external registry, so the digest might change over time.